core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -653,15 +657,19 @@ private RexNode simplifyUnaryPlus(RexCall call, RexUnknownAs unknownAs) {
     return simplify(call.getOperands().get(0), unknownAs);
   }
 
-  private RexNode simplifyIs(RexCall call, RexUnknownAs unknownAs) {
+  private @Nonnull RexNode simplifyIs(RexCall call, RexUnknownAs unknownAs) {
```

Comment:
Should we use `nonnull` for arguments and return values by default? In that case, there's no point in adding `@Nonnull` annotation.

Reply:
I added `@Nonnull` because most `simplifyXxx` methods return null if they are not able to simplify, whereas this an exception.

If we were starting from scratch, I agree with you, we would make `@Nonnull` the default (or better, use Kotlin where all types are non-null unless followed by `?`) but I don't think it's worth it to change the code to systematically declare whether each parameter or return is nullable.

Reply:
I have the same feeling too, `@Nonnull` should be the default.